controller/recurring_job_controller.go

```diff
@@ -175,6 +175,15 @@ func (control *RecurringJobController) syncRecurringJob(key string) (err error)
 			return err
 		}
 		log.Debug("Cannot find recurring job, may have been deleted")
+		volumes, err := control.ds.ListVolumes()
+		if err != nil {
+			return err
+		}
+		for _, vol := range volumes {
+			if err := control.detachVolumeAutoAttachedByRecurringJob(name, vol); err != nil {
```

Comment:
Wait... The detachment function in the recurring job is not removed yet. Do you still want to add this kind of duplicate handling?
https://github.com/longhorn/longhorn-manager/blob/ac190f908d33a8c0f135167d73c3aed904f16d62/app/recurring_job.go#L260

Reply:
After talking with Chin-Ya, I think the risk of this detachment logic is acceptable. It may be triggered only when the recurring job is gone but the pod does not handle the cleanup gracefully.
After adding a comment here, this PR is good to go.

Reply:
Agreed and thanks for adding the comments for the later revisit.